src/app/features/search-bar/search-bar.component.ts

```diff
@@ -0,0 +1,252 @@
+} from '@angular/core';
+import { T } from '../../t.const';
+import { FormControl } from '@angular/forms';
+import { BehaviorSubject, combineLatest, from, Observable } from 'rxjs';
+import {
+  tap,
+  map,
+  debounceTime,
+  switchMap,
+  withLatestFrom,
+  startWith,
+} from 'rxjs/operators';
+import { TaskService } from '../tasks/task.service';
+import { Router } from '@angular/router';
+import { TODAY_TAG } from '../tag/tag.const';
+import { Project } from '../project/project.model';
+import { Tag } from '../tag/tag.model';
+import { ProjectService } from '../project/project.service';
+import { TagService } from '../tag/tag.service';
+import { Task } from '../tasks/task.model';
+import { blendInOutAnimation } from 'src/app/ui/animations/blend-in-out.ani';
+import { AnimationEvent } from '@angular/animations';
+import { SearchItem, SearchQueryParams } from './search-bar.model';
+import { getWorklogStr } from '../../util/get-work-log-str';
+import { devError } from 'src/app/util/dev-error';
+
+@Component({
+  selector: 'search-bar',
+  templateUrl: './search-bar.component.html',
+  styleUrls: ['./search-bar.component.scss'],
+  animations: [blendInOutAnimation],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class SearchBarComponent implements AfterViewInit, OnDestroy {
+  @Input() isElevated: boolean = false;
+  @Output() blurred: EventEmitter<any> = new EventEmitter();
+
+  @ViewChild('inputEl') inputEl!: ElementRef;
+  @ViewChild('searchForm') searchForm!: ElementRef;
+
+  T: typeof T = T;
+  isLoading$: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(true);
+  taskSuggestionsCtrl: FormControl = new FormControl();
+  filteredIssueSuggestions$: Observable<SearchItem[]> = new Observable();
+  isArchivedTasks: boolean = false;
+  isArchivedTasks$: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(false);
+  tooManyResults: boolean = false;
+
+  private _attachKeyDownHandlerTimeout?: number;
+  private _tasks: Task[] = [];
+  private _projects: Project[] = [];
+  private _tags: Tag[] = [];
+  private searchableItems$: Observable<SearchItem[]> = this.isArchivedTasks$.pipe(
+    switchMap((isArchivedTasks) => this._loadTasks$(isArchivedTasks)),
+    withLatestFrom(this._projectService.list$, this._tagService.tags$),
+    tap(([tasks, projects, tags]) => this._keepForLater(tasks, projects, tags)),
+    map(([tasks]) => this._mapTasksToSearchItems(tasks)),
+  );
+
+  constructor(
+    private _taskService: TaskService,
+    private _projectService: ProjectService,
+    private _tagService: TagService,
+    private _router: Router,
+  ) {}
+
+  private _loadTasks$(isArchivedTasks: boolean): Observable<Task[]> {
+    return !isArchivedTasks
+      ? this._taskService.allTasks$
+      : from(this._taskService.getArchivedTasks());
+  }
+
+  private _keepForLater(tasks: Task[], projects: Project[], tags: Tag[]): void {
+    this._tasks = tasks;
+    this._projects = projects;
+    this._tags = tags;
+  }
```

Comment:
I need tags for the icon generation for the search list. This one could be passed through about 3 methods to finally get where it is needed. As you prefer... 
Projects and tasks are needed later while handling onOptionSelected.
I hope it's Ok to "stash" these variables away for that occasion. Seemed to me the cleanest way and this time, to my understanding, they will be set without doubt.

Reply:
It's cleaner to do this in a new subscriptiion rather than as a side effect here. If you have multiple subscriptions to searchableItems$ this get's executed for every single one even though it is just needed once.